core/src/main/scala/chisel3/experimental/hierarchy/core/Lookupable.scala

```diff
@@ -13,13 +13,31 @@ import chisel3.internal.firrtl.ir.{Arg, ILit, Index, LitIndex, ModuleIO, Slot, U
 import chisel3.internal.{throwException, Builder, ViewParent}
 import chisel3.internal.binding.{AggregateViewBinding, ChildBinding, CrossModuleBinding, ViewBinding, ViewWriteability}
 
-/** Represents lookup typeclass to determine how a value accessed from an original IsInstantiable
-  *   should be tweaked to return the Instance's version
-  * Sealed.
+/** Typeclass used to recontextualize values from an original Definition to an Instance
+  *
+  * Implementations for Chisel types and other common Scala types are provided in the companion object.
+  * Users can define Lookupable for their own types.
+  *
+  * @example {{{
+  * case class Foo[T <: Data](name: String, data: T)
+  * object Foo {
+  *   implicit def lookupable[T <: Data]: Lookupable.Simple[Foo[T]] =
+  *     Lookupable.product2[Foo[T], String, T](
+  *       x => (x.name, x.data),
+  *       Foo.apply
+  *     )
+  * }
+  * }}}
   */
 @implicitNotFound(
-  "@public is only legal within a class or trait marked @instantiable, and only on vals of type" +
-    " Data, BaseModule, MemBase, IsInstantiable, IsLookupable, or Instance[_], or in an Iterable, Option, Either, or Tuple2"
+  "@public is only legal within a class or trait marked @instantiable, and only on vals of types" +
```

Comment:
i thought user can no longer mark things @instantiable...? Or they can still mark Modules and such as @instantiable to get the @public API?

Reply:
They can (and should) still mark `Modules` as `@instantiable`, but cannot[^1] mark user-defined types `@instantiable`.

[^1]: Or really _should not_, I can't figure out how to enforce it on the `@instantiable` macro because it runs before type inference. It's only enforced at the point of marking a val of the user-define type `@public`